### PR TITLE
ESP32: Update README of esp32 supported examples

### LINKED ISSUE #7185: ESP32: Update README for all esp32 examples.
#### Problem
The README under CHIP esp32 examples is outdated.

 #### Proposed Solution
Update all README.md for esp32.

### PR DESCRIPTION
**Pull request by sweetymhaiske**
#### Problem
  * README under CHIP esp32 examples are outdated.
  * Fixes #7185 
  
  #### Change overview
1.  README.md files for all esp32 examples are updated.
2. Set latest commit id for esp-idf release/v4.2 ( d9ec7df3931bea46292f015a99d19a28670bd561 ) for GitHub CI.
  
  #### Testing
  * Updated README based on current commissioning and cluster control.

### CODE REVIEW

File: examples/all-clusters-app/esp32/README.md [hunk 1]
```diff
@@ -55,42 +56,29 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
...
         $ cd {path-to-connectedhomeip}
+
+    To download and install packages.
+
         $ source ./scripts/bootstrap.sh
         $ source ./scripts/activate.sh
-        $ cd {path-to-connectedhomeip-examples}
 
-If packages are already installed then simply activate it.
+    If packages are already installed then simply activate it.
```
Comment (msandstedt) [inline examples/all-clusters-app/esp32/README.md:69]:
```suggestion
    If packages are already installed then simply activate them.
```

File: examples/all-clusters-app/esp32/README.md [hunk 2]
```diff
@@ -55,42 +56,29 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
...
-        $ ./install.sh
-        $ . ./export.sh
-        $ cd {path-to-connectedhomeip}
         $ source ./scripts/activate.sh
-        $ cd {path-to-connectedhomeip-examples}
 
 -   Configuration Options
 
-        To choose from the different configuration options, run menuconfig
+To choose from the different configuration options, run menuconfig
```
Comment (msandstedt) [inline examples/all-clusters-app/esp32/README.md:75]:
```suggestion
To choose from the different configuration options, run menuconfig.
```

File: examples/all-clusters-app/esp32/README.md [hunk 3]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
 
           $ idf.py monitor ESPPORT=/dev/tty.SLAB_USBtoUART
 
-## Using the Echo Server
+## Commissioning and cluster control
+
+Commissioning can be carried out using WiFi, BLE or ByPass.
 
-There are two ways to use the Echo Server running on the device.
+1.  Set the `Rendezvous Mode` for commissioning using menuconfig, the default
```
Comment (msandstedt) [inline examples/all-clusters-app/esp32/README.md:115]:
```suggestion
1.  Set the `Rendezvous Mode` for commissioning using menuconfig; the default
```

File: examples/all-clusters-app/esp32/README.md [hunk 4]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
 
           $ idf.py monitor ESPPORT=/dev/tty.SLAB_USBtoUART
 
-## Using the Echo Server
+## Commissioning and cluster control
+
+Commissioning can be carried out using WiFi, BLE or ByPass.
```
Comment (msandstedt) [inline examples/all-clusters-app/esp32/README.md:113]:
```suggestion
Commissioning can be carried out using WiFi, BLE or Bypass.
```

File: examples/all-clusters-app/esp32/README.md [hunk 5]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
...
+    Rendezvous mode is BLE.
 
-### Connect the ESP32 to a 2.4GHz Network of your choice
+         $ idf.py menuconfig
 
-1.  If the `WiFi Station Options` mentioned above are populated through
-    menuconfig, then ESP32 connects to the AP with those credentials (STA mode).
+Select the Rendezvous Mode via `Demo -> Rendezvous Mode`. If Rendezvous Mode is
+ByPass then set the credentials of WiFi Network i.e SSID and Password from
+menuconfig.
```
Comment (msandstedt) [inline examples/all-clusters-app/esp32/README.md:122]:
```suggestion
Bypass then set the credentials of the WiFi Network (i.e. SSID and Password from
menuconfig).
```

File: examples/all-clusters-app/esp32/README.md [hunk 6]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
...
+`idf.py menuconfig -> Component config -> CHIP Device Layer -> WiFi Station Options`
 
 2.  Now flash the device with the same command as before. (Use the right `/dev`
     device)
 
           $ idf.py flash monitor ESPPORT=/dev/tty.SLAB_USBtoUART
 
-3.  The device should boot up and connect to your network. When that happens you
+3.  The device should boot up and when device connects to your network then you
     will see a log like this in the monitor.
```
Comment (msandstedt) [inline examples/all-clusters-app/esp32/README.md:132]:
```suggestion
3.  The device should boot up.  When the device connects to your network, you
    will see a log like this in the monitor.
```

File: examples/lock-app/esp32/README.md [hunk 7]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
 
           $ idf.py monitor ESPPORT=/dev/ttyUSB0
 
-## Using the Echo Server
+## Commissioning and cluster control
+
+Commissioning can be carried out using WiFi, BLE or ByPass.
```
Comment (msandstedt) [inline examples/lock-app/esp32/README.md:89]:
```suggestion
Commissioning can be carried out using WiFi, BLE or Bypass.
```

File: examples/lock-app/esp32/README.md [hunk 8]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
...
+
+1.  Set the `Rendezvous Mode` for commissioning using menuconfig, the default
+    Rendezvous mode is BLE.
+
+         $ idf.py menuconfig
 
-### Connect the ESP32 to a 2.4GHz Network of your choice
+Select the Rendezvous Mode via `Demo -> Rendezvous Mode`. If Rendezvous Mode is
+ByPass then set the credentials of WiFi Network i.e SSID and Password from
```
Comment (msandstedt) [inline examples/lock-app/esp32/README.md:97]:
```suggestion
Bypass then set the credentials of WiFi Network i.e SSID and Password from
```

File: examples/lock-app/esp32/README.md [hunk 9]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
...
 
     $ idf.py erase_flash ESPPORT=/dev/ttyUSB0
 
-The demo application supports on-off cluster.
+### Setting up python controller
+
+Once ESP32 is up and running, we need to setup device controller to perform
+commissioning and cluster control.
+
+-   Set up python controller
```
Comment (msandstedt) [inline examples/lock-app/esp32/README.md:133]:
```suggestion
-   Set up python controller.
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 10]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
 
           $ idf.py monitor ESPPORT=/dev/ttyUSB0
 
-## Using the Echo Server
+## Commissioning and cluster control
+
+Commissioning can be carried out using WiFi, BLE or ByPass.
```
Comment (msandstedt) [inline examples/temperature-measurement-app/esp32/README.md:89]:
```suggestion
Commissioning can be carried out using WiFi, BLE or Bypass.
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 11]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
...
+1.  Set the `Rendezvous Mode` for commissioning using menuconfig, the default
+    Rendezvous mode is BLE.
 
-### Connect the ESP32 to a 2.4GHz Network of your choice
+         $ idf.py menuconfig
 
-1.  If the `WiFi Station Options` mentioned above are populated through
-    menuconfig, then ESP32 connects to the AP with those credentials (STA mode).
+Select the Rendezvous Mode via `Demo -> Rendezvous Mode`. If Rendezvous Mode is
+ByPass then set the credentials of WiFi Network i.e SSID and Password from
```
Comment (msandstedt) [inline examples/temperature-measurement-app/esp32/README.md:97]:
```suggestion
Bypass then set the credentials of WiFi Network i.e SSID and Password from
```

**Review by msandstedt — APPROVED** [review]
I had some minor editorial comments.  Feel to incorporate these if you choose.

The content though looks good.

File: examples/all-clusters-app/esp32/README.md [hunk 12]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
...
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
           $ idf.py erase_flash ESPPORT=/dev/tty.SLAB_USBtoUART
 
-### Use the ESP32's Network
+### Setting up python controller
 
-Alternatively, you can connect to the ESP32's Soft-AP directly.
+Once ESP32 is up and running, we need to setup device controller to perform
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:154]:
```suggestion
Once ESP32 is up and running, we need to set up a device controller to perform
```

File: examples/all-clusters-app/esp32/README.md [hunk 13]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
...
     will see a log like this in the monitor.
 
           I (5524) chip[DL]: SYSTEM_EVENT_STA_GOT_IP
           I (5524) chip[DL]: IPv4 address changed on WiFi station interface: <IP_ADDRESS>...
 
-    Note: If you are using the M5Stack, the screen will display the server's IP
-    Address if it successfully connects to the configured 2.4GHz Network.
-
-4.  Use
+4.  Controller used
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:137]:
```suggestion
4.  Use
```

File: examples/all-clusters-app/esp32/README.md [hunk 14]
```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
...
 
-3.  Use
-    [standalone chip-tool](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
-    or
-    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin)
-    to communicate with the device.
+-   Execute the controller.
+
+           $ source ./out/python_env/bin/activate
+           $ sudo chip-device-ctrl
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:165]: This sudo bit is ... not desirable.  Why are we asking people to do that?

File: examples/lock-app/esp32/README.md [hunk 15]
```diff
@@ -30,24 +32,19 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
...
         $ cd {path-to-connectedhomeip}
+
+    To download and install packages.
+
         $ source ./scripts/bootstrap.sh
         $ source ./scripts/activate.sh
-        $ cd {path-to-connectedhomeip-examples}
 
-If packages are already installed then simply activate it.
+    If packages are already installed then simply activate it.
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/README.md:45]:
```suggestion
    If packages are already installed then simply activate them.
```

File: examples/lock-app/esp32/README.md [hunk 16]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
...
     will see a log like this in the monitor.
 
           I (5524) chip[DL]: SYSTEM_EVENT_STA_GOT_IP
           I (5524) chip[DL]: IPv4 address changed on WiFi station interface: <IP_ADDRESS>...
 
-    Note: If you are using the M5Stack, the screen will display the server's IP
-    Address if it successfully connects to the configured 2.4GHz Network.
-
-4.  Use
+4.  Controller used
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/README.md:113]:
```suggestion
4.  Use
```

File: examples/lock-app/esp32/README.md [hunk 17]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
...
-    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin)
+    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin/CHIPTool)
+    or
+    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
     to communicate with the device.
 
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
     $ idf.py erase_flash ESPPORT=/dev/ttyUSB0
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/README.md:126]: This needs the right ESPPORT value.

File: examples/lock-app/esp32/README.md [hunk 18]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
 
           $ idf.py monitor ESPPORT=/dev/ttyUSB0
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/README.md:85]: Needs right ESPPORT value?

File: examples/lock-app/esp32/README.md [hunk 19]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
...
 
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
     $ idf.py erase_flash ESPPORT=/dev/ttyUSB0
 
-The demo application supports on-off cluster.
+### Setting up python controller
+
+Once ESP32 is up and running, we need to setup device controller to perform
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/README.md:130]:
```suggestion
Once ESP32 is up and running, we need to set up a device controller to perform
```

File: examples/lock-app/esp32/README.md [hunk 20]
```diff
@@ -95,36 +84,102 @@ If packages are already installed then simply activate it.
...
+
+-   Set up python controller
+
+           $ cd {path-to-connectedhomeip}
+           $ ./scripts/build_python.sh -m platform
+
+-   Execute the controller.
+
+           $ source ./out/python_env/bin/activate
+           $ sudo chip-device-ctrl
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/README.md:141]: Again, why sudo?  Shouldn't need sudo.

File: examples/persistent-storage/esp32/README.md [hunk 21]
```diff
@@ -61,24 +61,19 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
...
         $ cd {path-to-connectedhomeip}
+
+    To download and install packages.
+
         $ source ./scripts/bootstrap.sh
         $ source ./scripts/activate.sh
-        $ cd {path-to-connectedhomeip-examples}
 
-If packages are already installed then simply activate it.
+    If packages are already installed then simply activate it.
```
Comment (bzbarsky-apple) [inline examples/persistent-storage/esp32/README.md:74]:
```suggestion
    If packages are already installed then simply activate them.
```

File: examples/pigweed-app/esp32/README.md [hunk 22]
```diff
@@ -64,18 +64,15 @@ To download and install packages.
...
         $ cd {path-to-connectedhomeip}
+
+    To download and install packages.
+
         $ source ./scripts/bootstrap.sh
         $ source ./scripts/activate.sh
-        $ cd {path-to-connectedhomeip-examples}
 
-If packages are already installed then simply activate it.
+    If packages are already installed then simply activate it.
```
Comment (bzbarsky-apple) [inline examples/pigweed-app/esp32/README.md:73]:
```suggestion
    If packages are already installed then simply activate them.
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 23]
```diff
@@ -1,5 +1,7 @@
 # Temperature Sensor Example
 
+This example mainly focus on Optimization.
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:3]:
```suggestion
This example is meant to represent a minimal-sized application.
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 24]
```diff
@@ -30,24 +32,19 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
...
         $ cd {path-to-connectedhomeip}
+
+    To download and install packages.
+
         $ source ./scripts/bootstrap.sh
         $ source ./scripts/activate.sh
-        $ cd {path-to-connectedhomeip-examples}
 
-If packages are already installed then simply activate it.
+    If packages are already installed then simply activate it.
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:45]:
```suggestion
    If packages are already installed then simply activate them.
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 25]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
 
           $ idf.py monitor ESPPORT=/dev/ttyUSB0
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:85]: Fix the ESPPORT?

File: examples/temperature-measurement-app/esp32/README.md [hunk 26]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
...
     will see a log like this in the monitor.
 
           I (5524) chip[DL]: SYSTEM_EVENT_STA_GOT_IP
           I (5524) chip[DL]: IPv4 address changed on WiFi station interface: <IP_ADDRESS>...
 
-    Note: If you are using the M5Stack, the screen will display the server's IP
-    Address if it successfully connects to the configured 2.4GHz Network.
-
-4.  Use
+4.  Controller used
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:113]:
```suggestion
4.  Use
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 27]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
...
-    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin)
+    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin/CHIPTool)
+    or
+    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
     to communicate with the device.
 
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
     $ idf.py erase_flash ESPPORT=/dev/ttyUSB0
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:126]: Fix ESPPORT?

File: examples/temperature-measurement-app/esp32/README.md [hunk 28]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
...
 
-The demo application supports temperaturemeasurement and basic cluster.
+### Setting up python controller
+
+Once ESP32 is up and running, we need to setup device controller to perform
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:130]:
```suggestion
Once ESP32 is up and running, we need to set up a device controller to perform
```

File: examples/temperature-measurement-app/esp32/README.md [hunk 29]
```diff
@@ -95,39 +84,92 @@ If packages are already installed then simply activate it.
...
+
+-   Set up python controller
+
+           $ cd {path-to-connectedhomeip}
+           $ ./scripts/build_python.sh -m platform
+
+-   Execute the controller.
+
+           $ source ./out/python_env/bin/activate
+           $ sudo chip-device-ctrl
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:141]: No sudo.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/README.md (hunk 1 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md]: Done.

File: examples/all-clusters-app/esp32/README.md (hunk 2 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md]: Done

**Comment by sweetymhaiske** [comment]
> I had some minor editorial comments. Feel to incorporate these if you choose.
> 
> The content though looks good.

I have updated it. Thanks

File: examples/all-clusters-app/esp32/README.md (hunk 14 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md]: Removed.

File: examples/lock-app/esp32/README.md (hunk 17 above)
Comment (sweetymhaiske) [reply examples/lock-app/esp32/README.md]: Fixed.

File: examples/temperature-measurement-app/esp32/README.md (hunk 25 above)
Comment (sweetymhaiske) [reply examples/temperature-measurement-app/esp32/README.md]: Fixed.

File: examples/temperature-measurement-app/esp32/README.md (hunk 29 above)
Comment (sweetymhaiske) [reply examples/temperature-measurement-app/esp32/README.md]: Yup done.

File: examples/lock-app/esp32/README.md (hunk 20 above)
Comment (sweetymhaiske) [reply examples/lock-app/esp32/README.md]: Yes Removed.

File: examples/all-clusters-app/esp32/README.md [hunk 30]
```diff
@@ -1,29 +1,30 @@
...
+-   [CHIP All Clusters Example](#chip-all-clusters-example)
     -   [Supported Devices](#supported-devices)
     -   [Building the Example Application](#building-the-example-application)
         -   [To build the application, follow these steps:](#to-build-the-application-follow-these-steps)
-    -   [Using the Echo Server](#using-the-echo-server)
-        -   [Connect the ESP32 to a 2.4GHz Network of your choice](#connect-the-esp32-to-a-24ghz-network-of-your-choice)
-        -   [Use the ESP32's Network](#use-the-esp32s-network)
+    -   [Commissioning and cluster control](#commissioning-and-cluster-control)
+        -   [Setting up python controller](#setting-up-python-controller)
+        -   [Commissioning esp32 device](#commissioning-using-ble)
```
[the remark this answers is not in the document]
Comment (dhrishi) [reply examples/all-clusters-app/esp32/README.md:14]: esp32 => ESP32 (at all places)

File: examples/all-clusters-app/esp32/README.md [hunk 31]
```diff
@@ -1,29 +1,30 @@
...
-The CHIP demo application is intended to work on two categories of ESP32
+The CHIP demo application is intended to work on three categories of ESP32
 devices: the
 [ESP32-DevKitC](https://www.espressif.com/en/products/hardware/esp32-devkitc/overview),
-and the [M5Stack](http://m5stack.com). On the [M5Stack](http://m5stack.com) this
-example displays a CHIP QRCode with the device's Soft-AP SSID encoded in the TLV
-section.
+the
+[ESP32-WROVER-KIT_V4.1](https://www.espressif.com/en/products/hardware/esp-wrover-kit/overview),
+and the [M5Stack](http://m5stack.com).
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:27]: Explicitly mention M5Stack Core. As M5Stack Core 2 is not supported yet.

File: examples/all-clusters-app/esp32/README.md [hunk 32]
```diff
@@ -120,64 +108,111 @@ If packages are already installed then simply activate it.
...
 
 2.  Now flash the device with the same command as before. (Use the right `/dev`
     device)
 
           $ idf.py flash monitor ESPPORT=/dev/tty.SLAB_USBtoUART
 
-3.  The device should boot up and connect to your network. When that happens you
-    will see a log like this in the monitor.
+3.  The device should boot up. When device connects to your network, you will
+    see a log like this in the monitor.
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:132]: in the monitor => on the device console?

File: examples/all-clusters-app/esp32/README.md [hunk 33]
```diff
@@ -120,64 +108,111 @@ If packages are already installed then simply activate it.
...
+    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
     to communicate with the device.
 
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
           $ idf.py erase_flash ESPPORT=/dev/tty.SLAB_USBtoUART
 
-### Use the ESP32's Network
+### Setting up python controller
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:152]: python controller -> Python Controller

File: examples/all-clusters-app/esp32/README.md [hunk 34]
```diff
@@ -120,64 +108,111 @@ If packages are already installed then simply activate it.
...
+
+### Note
 
-In addition to the echo server, this demo also supports controlling OnOff
-cluster (Server) attributes of an endpoint. For `ESP32-DevKitC` and
-`ESP32-WROVER-KIT_V4.1`, a GPIO (configurable through `STATUS_LED_GPIO_NUM` in
-`main/main.cpp`) is updated through the on/off/toggle commands from the
-`chip-tool`. For `M5Stack`, a virtual Green LED on the display is used for the
-same.
+This demo app illustrated controlling OnOff cluster (Server) attributes of an
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:210]: illustrates?

File: examples/lock-app/esp32/README.md [hunk 35]
```diff
@@ -1,5 +1,7 @@
 # CHIP ESP32 Lock Example
 
+This example demonstrate the mapping of OnOff cluster to lock/unlock logic.
```
Comment (dhrishi) [inline examples/lock-app/esp32/README.md:3]: demonstrates

File: examples/all-clusters-app/esp32/README.md [hunk 36]
```diff
@@ -120,64 +108,111 @@ If packages are already installed then simply activate it.
...
+
+         - chip-device-ctrl > resolve 0 135246
+
+### Cluster control
+
+-   After successful commissioning, use the OnOff cluster command to control the
+    OnOff attribute. This allows you to toggle a parameter implemented by the
+    device to be On or Off.
+
+    `chip-device-ctrl > zcl OnOff Off 135246 0 0`
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:206]: `0 0` => `1 0` for cluster commands (at all the places)

File: examples/temperature-measurement-app/esp32/README.md [hunk 37]
```diff
@@ -93,41 +82,102 @@ If packages are already installed then simply activate it.
...
+
+         - chip-device-ctrl > close-ble
+
+-   Resolve DNS-SD name and update address of the node in the device controller.
+
+         - chip-device-ctrl > resolve 0 135246
+
+### Cluster control
+
+-   The demo application supports temperaturemeasurement and basic cluster.
```
Comment (dhrishi) [inline examples/temperature-measurement-app/esp32/README.md:178]: temperature-measurement?

**Review by dhrishi — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/README.md [hunk 38]
```diff
@@ -1,29 +1,30 @@
...
--   [CHIP App Server Example](#chip-app-server-example)
+-   [CHIP All Clusters Example](#chip-all-clusters-example)
     -   [Supported Devices](#supported-devices)
     -   [Building the Example Application](#building-the-example-application)
         -   [To build the application, follow these steps:](#to-build-the-application-follow-these-steps)
-    -   [Using the Echo Server](#using-the-echo-server)
-        -   [Connect the ESP32 to a 2.4GHz Network of your choice](#connect-the-esp32-to-a-24ghz-network-of-your-choice)
-        -   [Use the ESP32's Network](#use-the-esp32s-network)
+    -   [Commissioning and cluster control](#commissioning-and-cluster-control)
+        -   [Setting up python controller](#setting-up-python-controller)
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:13]: Can we remove `To build the application, follow these steps:` from being a separate section? I don't think that a separate heading is required for that

**Review by dhrishi — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/README.md (hunk 38 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md]: Done

File: examples/temperature-measurement-app/esp32/README.md (hunk 37 above)
Comment (sweetymhaiske) [reply examples/temperature-measurement-app/esp32/README.md]: Renamed it as it is present for the python controller ZCL cluster.

File: integrations/docker/images/chip-build-esp32/Dockerfile [hunk 39]
```diff
@@ -9,6 +9,7 @@ RUN set -x \
     && cd /opt/espressif \
     && git clone --progress -b release/v4.2 https://github.com/espressif/esp-idf.git \
     && cd esp-idf \
+    && git reset --hard d9ec7df3931bea46292f015a99d19a28670bd561
```
Comment (dhrishi) [inline integrations/docker/images/chip-build-esp32/Dockerfile:12]: Hey @andy31415  We need to increment the dockerfile version after this PR is merged?

**Review by bzbarsky-apple — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

File: integrations/docker/images/chip-build-esp32/Dockerfile (hunk 39 above)
Comment (dhrishi) [reply integrations/docker/images/chip-build-esp32/Dockerfile:12]: I think we can do this after merge.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**